### PR TITLE
docs(docs): Add ADR on structured logging.

### PR DESCRIPTION
**Pull request by stevenj**
# Description

Adds an ADR about structured logging.

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

**Comment by stevenj** [comment]
> That sounds like a good idea. Are we going to have concrete specs for this later? For example, we said, `Date and Time *MUST* always be referenced to UTC and not Local Time.` But what exact format are we going to use? For example, we'd prefer `ISO 8601,` not `RFC 1123 `.

Yes we can.  The only reason I didn't is because I wasn't sure what would be easyacross all the languages.  For Rust we are using rfc_3339 (a subset of ISO 8601).
So we could say the time can be formatted in either `ISO 8601` or `RF 3339`.

File: docs/src/architecture/09_architecture_decisions/0008-structured-logging.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+## Decision
+
+* We will use structured logging for all log messages that are to be sent to a log collection system.
+* We will not use string formatting within structured log messages, but embed all data within fields in the log message.
+* All log levels *MUST* use structured logging.
+  * It is possible that **DEBUG** or **TRACE** level logs can selectively be enabled in production to help investigate issues.
+* Log text messages should be specific enough to identify the actual log messages when searching or summarizing logs.
+  * Avoid using the same log message multiple times in the same code.
+* As far as we are able, we should be consistent with field names.
+  * For example, we always use "error" for error values, not "error" or "err" or "e".
```
Comment (bkioshn) [inline docs/src/architecture/09_architecture_decisions/0008-structured-logging.md:33]: not "err" or "e".

File: docs/src/architecture/09_architecture_decisions/0008-structured-logging.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+* Log text messages should be specific enough to identify the actual log messages when searching or summarizing logs.
+Don't use the same log message multiple times in the same code.
+* As far as we are able, we should be consistent with field names.
+For example, we always use "error" for error values, not "error" or "err" or "e".
+Fields do not need to have the same name as the variable, what is important is the field name be unambiguous and consistent.
+* Logs *MUST* include the date and time of the event being logged.
+Date and Time *MUST* always be referenced to UTC, and not Local Time.
+* Logs *SHOULD*, if possible, include the filename, and line of the log message in the code.
+* It is *RECOMMENDED* that basic log information be pre-configured such that all log messages automatically include it.
+* API Endpoints *MUST* obfuscate sensitive information before logging, such as IP Addresses of the client.
```
[the remark this answers is not in the document]
Comment (stevenj) [reply docs/src/architecture/09_architecture_decisions/0008-structured-logging.md]: Clarified the wording as suggested.

File: docs/src/architecture/09_architecture_decisions/0008-structured-logging.md (hunk 1 above)
Comment (stevenj) [reply docs/src/architecture/09_architecture_decisions/0008-structured-logging.md]: Fixed

**Review by minikin — APPROVED** [review]

**Pull request merged**